File: modProcedureVariable.vb
Imports MySqlConnector
Imports System.Windows.Forms

Namespace Student_Assessment_System_with_Item_Analysis
    Public Module modProcedureVariable

        Public Function fncConnectToDatabase() As Boolean
            Try
                servername = "localhost"
                dbname = "studentassessment"
                username = "root"
                password = ""
                port = "3306"

                strConnection = $"SERVER={servername};DATABASE={dbname};UID={username};PWD={password};PORT={port};"

                conDbInformation = New MySqlConnection(strConnection)
                conDbInformation.Open()

                command = New MySqlCommand()
                command.Connection = conDbInformation
                command.CommandTimeout = 20 * 60

                Return True
            Catch ex As Exception
                MessageBox.Show("Database Connection Error: " & ex.Message)
                Return False
            End Try
        End Function

        Public Sub checkDatabaseConnection()
            If conDbInformation Is Nothing OrElse conDbInformation.State = ConnectionState.Closed Then
                fncConnectToDatabase()
            End If
        End Sub

    End Module
End Namespace

File: modProcedureVariable.cs
using MySql.Data.MySqlClient;
using System;
using System.Windows.Forms;

namespace Student_Assessment_System_with_Item_Analysis
{
    public static class modProcedureVariable
    {
        public static bool fncConnectToDatabase()
        {
            try
            {
                modVariables.servername = "localhost";
                modVariables.dbname = "studentasessment";
                modVariables.username = "root";
                modVariables.password = "";
                modVariables.port = "3306";

                modVariables.strConnection =
                    $"SERVER={modVariables.servername};DATABASE={modVariables.dbname};UID={modVariables.username};PWD={modVariables.password};PORT={modVariables.port};";

                modVariables.conDbInformation = new MySqlConnection(modVariables.strConnection);
                modVariables.conDbInformation.Open();

                modVariables.command = new MySqlCommand();
                modVariables.command.Connection = modVariables.conDbInformation;
                modVariables.command.CommandTimeout = 20 * 60;

                return true;
            }
            catch (Exception ex)
            {
                MessageBox.Show("Database Connection Error: " + ex.Message);
                return false;
            }
        }

        public static void checkDatabaseConnection()
        {
            if (modVariables.conDbInformation == null ||
                modVariables.conDbInformation.State == System.Data.ConnectionState.Closed)
            {
                fncConnectToDatabase();
            }
        }
    }
}
